python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -263,9 +262,10 @@ def _sample_pq_statistics(ir: Scan) -> dict[str, np.floating[T]]:
     import numpy as np
 
     # Use average total_uncompressed_size of three files
-    n_sample = min(3, len(ir.paths))
+    n_sample = 5  # TODO: Make this configurable
+    stride = max(1, int(len(ir.paths) / n_sample))
     metadata = plc.io.parquet_metadata.read_parquet_metadata(
-        plc.io.SourceInfo(random.sample(ir.paths, n_sample))
+        plc.io.SourceInfo(ir.paths[: stride * n_sample : stride])
```

Comment:
This "fix" is not related to the rest of the PR, but it made it easier to do A-B testing with the new fusion optimization.